--- a/hw/ip/prim/rtl/prim_fifo_sync_cnt.sv
+++ b/hw/ip/prim/rtl/prim_fifo_sync_cnt.sv
@@ -68,7 +68,7 @@
   // - when only one of the two 'wrap' pointers is wrapped: the maximum depth minus the difference
   //   of the real pointers.
   assign depth_o = full_o                         ? DepthW'(Depth) :
-                   wptr_wrap_msb == rptr_wrap_msb ? DepthW'(wptr_o) - DepthW'(rptr_o) :
+                   wptr_wrap_msb == rptr_wrap_msb ? DepthW'(wptr_o - rptr_o) :
                    DepthW'(Depth) - DepthW'(rptr_o) + DepthW'(wptr_o);
 
   if (Secure) begin : gen_secure_ptrs
