--- a/hw/ip/prim/rtl/prim_rom_adv.sv
+++ b/hw/ip/prim/rtl/prim_rom_adv.sv
@@ -33,11 +33,18 @@
     .rst_ni,
     .req_i,
     .addr_i,
-    .rvalid_o,
     .rdata_o,
     .cfg_i
   );
 
+  always_ff @(posedge clk_i or negedge rst_ni) begin
+    if (!rst_ni) begin
+      rvalid_o <= 1'b0;
+    end else begin
+      rvalid_o <= req_i;
+    end
+  end
+
   ////////////////
   // ASSERTIONS //
   ////////////////
