--- a/util/reggen/systemrdl_exporter.py
+++ b/util/reggen/systemrdl_exporter.py
@@ -121,7 +121,7 @@ class Window2Systemrdl:
     def export(self) -> systemrdl.component.Mem:
         rdl_mem_t = self.importer.create_mem_definition(self.inner.name)
         self.importer.assign_property(
-            rdl_mem_t, "memwidth", self.inner.size_in_bytes // self.inner.items
+            rdl_mem_t, "memwidth", (self.inner.size_in_bytes // self.inner.items) * 8
         )
         return self.importer.instantiate_mem(
             rdl_mem_t, self.inner.name, self.inner.offset, [self.inner.items]
