--- a/hw/ip/prim/rtl/prim_util_memload.svh
+++ b/hw/ip/prim/rtl/prim_util_memload.svh
@@ -15,7 +15,7 @@
  * - A parameter `MemInitFile` with a file path of a VMEM file to be loaded into
  *   the memory if not empty.
  *
- * Note this works with memories up to a maximum width of 312 bits. Should this maximum width be
+ * Note this works with memories up to a maximum width of 320 bits. Should this maximum width be
  * increased all of the `simutil_set_mem` and `simutil_get_mem` call sites must be found (e.g. using
  * git grep) and adjusted appropriately.
  */
@@ -33,9 +33,9 @@
   // Returns 1 (true) for success, 0 (false) for errors.
   export "DPI-C" function simutil_set_mem;
 
-  function int simutil_set_mem(input int index, input bit [311:0] val);
+  function int simutil_set_mem(input int index, input bit [319:0] val);
     int valid;
-    valid = Width > 312 || index >= Depth ? 0 : 1;
+    valid = Width > 320 || index >= Depth ? 0 : 1;
     if (valid == 1) mem[index] = val[Width-1:0];
     return valid;
   endfunction
@@ -43,9 +43,9 @@
   // Function for getting a specific element in |mem|
   export "DPI-C" function simutil_get_mem;
 
-  function int simutil_get_mem(input int index, output bit [311:0] val);
+  function int simutil_get_mem(input int index, output bit [319:0] val);
     int valid;
-    valid = Width > 312 || index >= Depth ? 0 : 1;
+    valid = Width > 320 || index >= Depth ? 0 : 1;
     if (valid == 1) begin
       val = 0;
       val[Width-1:0] = mem[index];
